diff --git a/index.d.ts b/index.d.ts
index adcfd85c19896cb198932bb6a359beaaf366d92a..74a073932b00e52b923b6701ccbbc283041e552a 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -21,6 +21,46 @@ declare namespace facebook {
     > = TEvent extends "xfbl.render" ? () => void
         : (response: StatusResponse) => void;
 
+    interface UserResponse {
+        id: number;
+        about?: string;
+        age_range?: AgeRange;
+        birthday?: string;
+        education?: EducationExperience;
+        picture?: string;
+        email?: string;
+        favorite_athletes?: Experience[];
+        favorite_teams?: Experience[];
+        first_name?: string;
+        gender?: string;
+        hometown?: string;
+        inspirational_people?: Experience[];
+        install_type?: string;
+        is_guest_user?: boolean;
+        languages?: Experience[];
+        last_name?: string;
+        link?: string;
+        location?: Page;
+        meeting_for?: string[];
+        middle_name?: string;
+        name?: string;
+        name_format?: string;
+        payment_pricepoints?: PaymentPricepoints;
+        name_political?: string;
+        profile_pic?: string;
+        quotes?: string;
+        relationship_status?: string;
+        religion?: string;
+        shared_login_upgrade_required_by?: any;
+        short_name?: string;
+        significant_other?: User;
+        sports?: Experience[];
+        supports_donate_button_in_live_video?: boolean;
+        token_for_business?: string;
+        video_upload_limits?: VideoUploadLimits;
+        website?: string;
+    }
+
     type UserField =
         | "id"
         | "about"
@@ -104,7 +144,7 @@ declare namespace facebook {
                 gender?: TParam extends "gender" ? string : never | undefined | undefined;
                 hometown?: TParam extends "hometown" ? Page : never | undefined | undefined;
                 inspirational_people?: TParam extends "inspirational_people" ? Experience[]
-                    : never | undefined | undefined;
+                : never | undefined | undefined;
                 install_type?: TParam extends "install_type" ? any : never | undefined | undefined;
                 is_guest_user?: TParam extends "is_guest_user" ? boolean : never | undefined | undefined;
                 languages?: TParam extends "languages" ? Experience[] : never | undefined | undefined;
@@ -116,21 +156,21 @@ declare namespace facebook {
                 name?: TParam extends "name" ? string : never | undefined | undefined;
                 name_format?: TParam extends "name_format" ? string : never | undefined | undefined;
                 payment_pricepoints?: TParam extends "payment_pricepoints" ? PaymentPricepoints
-                    : never | undefined | undefined;
+                : never | undefined | undefined;
                 name_political?: TParam extends "political" ? string : never | undefined | undefined;
                 profile_pic?: TParam extends "profile_pic" ? string : never | undefined | undefined;
                 quotes?: TParam extends "quotes" ? string : never | undefined | undefined;
                 relationship_status?: TParam extends "relationship_status" ? string : never | undefined | undefined;
                 religion?: TParam extends "religion" ? string : never | undefined | undefined;
                 shared_login_upgrade_required_by?: TParam extends "shared_login_upgrade_required_by" ? any
-                    : never | undefined | undefined;
+                : never | undefined | undefined;
                 short_name?: TParam extends "short_name" ? any : never | undefined | undefined;
                 significant_other?: TParam extends "significant_other" ? User : never | undefined | undefined;
                 sports?: TParam extends "sports" ? Experience[] : never | undefined | undefined;
                 supports_donate_button_in_live_video?: TParam extends "supports_donate_button_in_live_video" ? boolean
-                    : never | undefined | undefined;
+                : never | undefined | undefined;
                 token_for_business?: TParam extends "token_for_business" ? VideoUploadLimits
-                    : never | undefined | undefined;
+                : never | undefined | undefined;
                 video_upload_limits?: TParam extends "video_upload_limits" ? string : never | undefined | undefined;
                 website?: TParam extends "website" ? string : never | undefined | undefined;
             }) => void,
@@ -379,10 +419,10 @@ declare namespace facebook {
         data?: string | undefined;
         exclude_ids?: string[] | undefined;
         filters?:
-            | "app_users"
-            | "app_non_users"
-            | Array<{ name: string; user_ids: string[] }>
-            | undefined;
+        | "app_users"
+        | "app_non_users"
+        | Array<{ name: string; user_ids: string[] }>
+        | undefined;
         max_recipients?: number | undefined;
         object_id?: string | undefined;
         suggestions?: string[] | undefined;
@@ -428,12 +468,12 @@ declare namespace facebook {
         display: "popup";
         method: "create_offer";
         objective:
-            | "APP_INSTALLS"
-            | "CONVERSIONS"
-            | "LINK_CLICKS"
-            | "OFFER_CLAIMS"
-            | "PRODUCT_CATALOG_SALES"
-            | "STORE_VISITS";
+        | "APP_INSTALLS"
+        | "CONVERSIONS"
+        | "LINK_CLICKS"
+        | "OFFER_CLAIMS"
+        | "PRODUCT_CATALOG_SALES"
+        | "STORE_VISITS";
         page_id: string;
     }
 
@@ -547,5 +587,5 @@ declare namespace facebook {
         success: boolean;
     }
 
-    interface CollectionAdsDialogResponse extends InstantExperiencesAdsDialogResponse {}
+    interface CollectionAdsDialogResponse extends InstantExperiencesAdsDialogResponse { }
 }
